packages/next/src/client/components/router-reducer/reducers/navigate-reducer.ts

```diff
@@ -127,45 +127,37 @@ function navigateReducer_noPPR(
     return handleExternalUrl(state, mutable, url.toString(), pendingPush)
   }
 
-  const prefetchCacheKey = createPrefetchCacheKey(url, state.nextUrl)
-  let prefetchValues = state.prefetchCache.get(prefetchCacheKey)
+  const prefetchCacheKey = createPrefetchCacheKey(url)
```

Comment:
This could use more explanation because it's quite subtle.

The reason to check the less specific value first is because it's only added to the cache if the server has omitted Next-Url from the the Vary key.

I would structure it a bit differently so you don't even compute the "interception cache key" unless the first check fails.